api/src/main/java/io/grpc/Metadata.java

```diff
@@ -706,14 +815,36 @@ public String toString() {
      * @return a parsed instance of type T
      */
     abstract T parseBytes(byte[] serialized);
+
+    /**
+     * @return whether this key should be serialized to bytes lazily.
+     */
+    boolean serializeToStreams() {
```

Comment:
What about `isSerializationLazy()` or `serializableToStreams()` or `serializesToStreams()`? The current name looks like "serialize to streams now and return true if successful".

Reply:
Makeses sense. I used "serializesToStreams", though I was considering "shouldSerializeToStreams". lmk if you have a preference.